File: sabq app ios/sabq/Components/AIImageBadge.swift
import SwiftUI

/// AI-generated image disclosure badge.
/// Matches the web's `ImageWithCaption.tsx` (`bg-primary/90`).
struct AIImageBadge: View {
    let model: String?
    private let sizeScale: CGFloat

    /// Drives re-render when the user picks a new accent in Settings.
    @AppStorage("appAccent") private var accentRaw: String = AppAccent.blue.rawValue
    @Environment(\.colorScheme) private var colorScheme

    init(model: String? = nil, sizeScale: CGFloat = 1) {
        self.model = model
        self.sizeScale = sizeScale
    }

    var body: some View {
        HStack(spacing: 5 * sizeScale) {
            Text("مولدة بالذكاء الاصطناعي")
                .font(SabqFonts.app(size: 11 * sizeScale, weight: .semibold))
                .lineLimit(1)
                .minimumScaleFactor(0.7)
                .allowsTightening(true)
            Image(systemName: "sparkles")
                .font(SabqFonts.app(size: 11 * sizeScale, weight: .bold))
        }
        // Icon on the physical right, Arabic text flows inward.
        .environment(\.layoutDirection, .leftToRight)
        .foregroundStyle(.white)
        .padding(.horizontal, 10 * sizeScale)
        .padding(.vertical, 5 * sizeScale)
        .background(
            Capsule().fill(badgeFill)
        )
        .overlay(
            Capsule().stroke(.white.opacity(colorScheme == .dark ? 0.22 : 0.30), lineWidth: 0.5)
        )
        .shadow(color: .black.opacity(colorScheme == .dark ? 0.45 : 0.25), radius: 6, x: 0, y: 2)
        .accessibilityLabel("مولدة بالذكاء الاصطناعي")
    }

    /// Same token as the web `bg-primary/90` — follows accent picker + dark mode.
    private var badgeFill: Color {
        let accent = AppAccent(rawValue: accentRaw) ?? .blue
        let base = colorScheme == .dark ? accent.darkColor : accent.color
        return base.opacity(0.95)
    }
}

/// Physical screen corner — not semantic leading/trailing (immune to RTL flip).
enum AIImageBadgeCorner {
    /// Physical top-left — same side as `.primaryAction` toolbar (like / save / share) in RTL.
    case topLeading
    /// Physical top-right — feed cards / web hero.
    case topTrailing
}

/// Pins the badge to a fixed physical corner of the image bounds.
struct AIImageBadgeOverlay: View {
    let model: String?
    let inset: CGFloat
    let sizeScale: CGFloat
    let corner: AIImageBadgeCorner

    init(
        model: String? = nil,
        inset: CGFloat = 12,
        sizeScale: CGFloat = 1,
        corner: AIImageBadgeCorner = .topTrailing
    ) {
        self.model = model
        self.inset = inset
        self.sizeScale = sizeScale
        self.corner = corner
    }

    var body: some View {
        // Color.clear expands to the full overlay; ZStack alignment pins the badge.
        // GeometryReader was centering the badge in the hero — avoid it here.
        ZStack(alignment: zStackAlignment) {
            Color.clear
            AIImageBadge(model: model, sizeScale: sizeScale)
                .padding(.top, inset)
                .padding(.leading, corner == .topLeading ? inset : 0)
                .padding(.trailing, corner == .topTrailing ? inset : 0)
        }
        .environment(\.layoutDirection, .leftToRight)
        .allowsHitTesting(false)
    }

    private var zStackAlignment: Alignment {
        corner == .topLeading ? .topLeading : .topTrailing
    }
}

extension View {
    func aiImageBadgeOverlay(
        isVisible: Bool,
        model: String? = nil,
        inset: CGFloat = 12,
        sizeScale: CGFloat = 1,
        corner: AIImageBadgeCorner = .topTrailing
    ) -> some View {
        overlay {
            if isVisible {
                AIImageBadgeOverlay(
                    model: model,
                    inset: inset,
                    sizeScale: sizeScale,
                    corner: corner
                )
            }
        }
    }
}

File: android-native/app/src/main/kotlin/com/sabq/smart/ui/components/AIImageBadge.kt
package com.sabq.smart.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalLayoutDirection
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.LayoutDirection
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.CompositionLocalProvider
import com.sabq.smart.ui.theme.SabqTheme

/**
 * AI-generated image disclosure badge — pixel-port of iOS
 * `AIImageBadge` (Components/AIImageBadge.swift). Pill with accent fill,
 * white text "مولدة بالذكاء الاصطناعي" + trailing sparkles icon.
 *
 * Visual notes from iOS:
 * - Capsule fill: accent colour @ 0.95 opacity (NOT a dark media scrim)
 * - Border: white @ 0.30 light / 0.22 dark
 * - Shadow: black @ 0.25 light / 0.45 dark, radius 6 dp, y=2
 * - Internal LTR layout direction so icon stays physically right of text
 *   even when the surrounding screen is RTL.
 * - Model name is intentionally not rendered (iOS captures it for
 *   accessibility / future analytics).
 */
@Composable
fun AIImageBadge(
    modifier: Modifier = Modifier,
    model: String? = null,
    sizeScale: Float = 1f,
) {
    val border = if (SabqTheme.colors.isDark) Color.White.copy(alpha = 0.22f)
                 else Color.White.copy(alpha = 0.30f)
    val shadowColor = if (SabqTheme.colors.isDark) Color.Black.copy(alpha = 0.45f)
                      else Color.Black.copy(alpha = 0.25f)
    val fill = SabqTheme.colors.primaryEnd.copy(alpha = 0.95f)

    @Suppress("UnusedReceiverParameter")
    val unused = model // kept for parity with iOS API surface

    CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Ltr) {
        Row(
            modifier = modifier
                .shadow(elevation = 6.dp, shape = CircleShape, ambientColor = shadowColor, spotColor = shadowColor)
                .clip(CircleShape)
                .background(fill, CircleShape)
                .border(width = 0.5.dp, color = border, shape = CircleShape)
                .padding(horizontal = (10 * sizeScale).dp, vertical = (5 * sizeScale).dp)
                .semantics { contentDescription = "مولدة بالذكاء الاصطناعي" },
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy((5 * sizeScale).dp),
        ) {
            Text(
                text = "مولدة بالذكاء الاصطناعي",
                style = TextStyle(
                    fontSize = (10 * sizeScale).sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White,
                ),
                maxLines = 1,
            )
            Icon(
                imageVector = Icons.Filled.AutoAwesome,
                contentDescription = null,
                tint = Color.White,
                modifier = Modifier.size((10 * sizeScale).dp),
            )
        }
    }
}

/** Physical screen corner — immune to RTL flips. */
enum class AIImageBadgeCorner { TopStart, TopEnd }

/**
 * Pins the AI badge to a fixed physical corner of an image. Use as an
 * overlay inside the same Box that contains the hero / thumbnail.
 * Mirrors iOS `AIImageBadgeOverlay`.
 */
@Composable
fun BoxScopedAIImageBadgeOverlay(
    isVisible: Boolean,
    modifier: Modifier = Modifier,
    model: String? = null,
    inset: Dp = 12.dp,
    sizeScale: Float = 1f,
    corner: AIImageBadgeCorner = AIImageBadgeCorner.TopEnd,
) {
    if (!isVisible) return
    // The badge itself enforces LTR. We use Box alignment here on the
    // physical screen (TopEnd ≈ physical top-right) so RTL pages don't
    // flip the corner.
    Box(modifier = modifier.padding(inset)) {
        CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Ltr) {
            AIImageBadge(model = model, sizeScale = sizeScale)
        }
    }
    // Helper marker — actual alignment is applied by the caller via
    // Modifier.align(...) below in the overlay site to choose corner.
    @Suppress("UNUSED_EXPRESSION") corner
}
